### PR TITLE
Add FasterRCNN improved weights

### LINKED ISSUE #5307: Improve the accuracy of Detection models by using SOTA recipes and primitives
### 🚀 The feature

Similar to #3995 but focus on Object Detection.

Here is the accuracy of `fasterrcnn_resnet50_fpn` on latest main:
```
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.369
0:  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.585
0:  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.396
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.212
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.403
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.482
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.307
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.485
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.509
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.317
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.544
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.649
```

Some of the recent ML techniques added for the SSD work can immediately improve the accuracy of the FasterRCNN models without any additional changes. Obviously much bigger improvements can be achieved by carefully tuning our training recipes to the needs of each model. Using the SSD data augmentation on `fasterrcnn_resnet50_fpn` we can improve the accuracy by 2.1 mAP:
```
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.390
0:  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.601
0:  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.422
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.238
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.424
0:  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.500
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.323
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.520
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.547
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.374
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.587
0:  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.681
```

### Motivation, pitch

Kick-off a Batteries Included phase 2 project that will focus on improving object detection. After adding the necessary primitives, create a new recipe that improves the accuracy of existing models and retrain them to offer better weights to the community.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by datumbox**
Fixes #5307

Adds new pre-trained weights for FasterRCNN + ResNet50 + FPN for the v2 variant with post-paper optimizations (no FrozenBN + c5 instead of p5 input on extra layers + heavier RPN/Box Heads with BNs). It improves the previous baseline by +9.7 mAP.

Trained with:
```
python -u run_with_submitit.py --ngpus 8 --nodes 4 --dataset coco --model fasterrcnn_resnet50_fpn_v2 \
--epochs 400 --lr-steps 352 384 --lr 0.1 --batch-size 2 --weight-decay 0.00004 --sync-bn \
--data-augmentation lsj 
```

Verified with:
```
torchrun --nproc_per_node=1 train.py --test-only --weights FasterRCNN_ResNet50_FPN_V2_Weights.COCO_V1 \
--model fasterrcnn_resnet50_fpn_v2 -b 1

IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.467
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.673
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.511
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.303
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.506
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.602
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.366
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.579
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.609
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.441
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.645
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.754
```

### CODE REVIEW

**Review by YosuaMichael — APPROVED** [review]
LGTM!

**Comment by santhoshnumberone** [comment]
@datumbox 
Does this mean that the current best pre trained model available on pytorch is `Faster R-CNN ResNet-50 FPN` version 2 with mAP of `46.7` surpassing the previous best `FCOS ResNet-50 FPN` of `39.2`?

Could you please provide the link to the `code` and the `pre-trained` of `Faster R-CNN ResNet-50 FPN` version 2

**Comment by datumbox** [comment]
@santhoshnumberone The FCOS was not retrained. It's likely we can do better with it as well if we review the training recipe but that will be reviewed on the future.

> Could you please provide the link to the code and the pre-trained of Faster R-CNN ResNet-50 FPN version 2

The description of this PR gives you all this info. Use the latest main of TorchVision.

**Comment by santhoshnumberone** [comment]
@datumbox off the topicCan't wait to see this [DINO: DETR with Improved DeNoising Anchor Boxes for End-to-End Object Detection](https://paperswithcode.com/paper/dino-detr-with-improved-denoising-anchor-1)

**Comment by santhoshnumberone** [comment]
@datumbox off the topic

Can't wait to see this [DINO: DETR with Improved DeNoising Anchor Boxes for End-to-End Object Detection](https://paperswithcode.com/paper/dino-detr-with-improved-denoising-anchor-1) [paper](https://arxiv.org/pdf/2203.03605v3.pdf) retrained by you in future
Screenshot 2022-05-11 at 4 56 39 PM

**Comment by datumbox** [comment]
@santhoshnumberone We have a dedicated issue for getting such requests at https://github.com/pytorch/vision/issues/2707

I strongly recommend commenting the same thing there so that you get visibility and consider it on the future. Having this discussion here is likely to be forgotten/lost.
